cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/hbase/coprocessor/ConsumerConfigCache.java

```diff
@@ -161,18 +169,27 @@ public synchronized void updateCache() {
         }
       }
     }
-
   }
 
   private void startRefreshThread() {
-    this.refreshThread = new Thread("queue-cache-refresh") {
+    refreshThread = new Thread("queue-cache-refresh") {
       @Override
       public void run() {
-        while (!isInterrupted()) {
+        int tableNotFoundCount = 0;
+        while (!isInterrupted() && tableNotFoundCount < TABLE_NOT_FOUND_MAX_RETRY) {
           updateConfig();
           long now = System.currentTimeMillis();
           if (now > (lastUpdated + configCacheUpdateFrequency)) {
-            updateCache();
+            try {
+              updateCache();
+              tableNotFoundCount = 0;
+            } catch (TableNotFoundException e) {
+              tableNotFoundCount++;
+              LOG.error("Queue config table not found: {}. Retries remaining {}",
```

Comment:
Maybe not log single retry failure in ERROR, but instead at INFO and log as ERROR when we stop trying?